### ISSUE TITLE
Error when input [d] on collision question

### ISSUE THREAD

**Issue by gerasiov** [technicalpickles/homesick#134]
```
$ homesick link
    conflict  /home/gq/.zshrc exists
Overwrite /home/gq/.zshrc? (enter "h" for help) [Ynaqdh] h
        Y - yes, overwrite
        n - no, do not overwrite
        a - all, overwrite this and all others
        q - quit, abort
        d - diff, show the differences between the old and the new
        h - help, show this help
Overwrite /home/gq/.zshrc? (enter "h" for help) [Ynaqdh] d
/usr/lib/ruby/vendor_ruby/homesick/utils.rb:153:in `block in collision_accepted?': undefined local variable or method `source' for #<Homesick::CLI:0x9a43748> (NameError)
	from /usr/lib/ruby/vendor_ruby/thor/shell/basic.rb:254:in `block in file_collision'
```

**Comment by hobarrera** [comment]
Also having this issue:

```
    conflict  /home/hugo/.gnupg/gpg.conf exists
Overwrite /home/hugo/.gnupg/gpg.conf? (enter "h" for help) [Ynaqdh] d
/usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/lib/homesick/utils.rb:153:in `block in collision_accepted?': undefined local variable or method `source' for #<Homesick::CLI:0x000000013540c0> (NameError)
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/shell/basic.rb:254:in `block in file_collision'
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/shell/basic.rb:237:in `loop'
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/shell/basic.rb:237:in `file_collision'
        from /usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/lib/homesick/utils.rb:153:in `collision_accepted?'
        from /usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/lib/homesick/actions/file_actions.rb:78:in `handle_symlink_action'
        from /usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/lib/homesick/actions/file_actions.rb:61:in `ln_s'
        from /usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/lib/homesick/utils.rb:201:in `block in symlink_each'
        from /usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/lib/homesick/utils.rb:188:in `block (2 levels) in each_file'
        from /usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/lib/homesick/utils.rb:162:in `each'
        from /usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/lib/homesick/utils.rb:162:in `block in each_file'
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions.rb:184:in `block in inside'
        from /usr/lib/ruby/2.2.0/fileutils.rb:127:in `chdir'
        from /usr/lib/ruby/2.2.0/fileutils.rb:127:in `cd'
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions.rb:184:in `inside'
        from /usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/lib/homesick/utils.rb:158:in `each_file'
        from /usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/lib/homesick/utils.rb:200:in `symlink_each'
        from /usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/lib/homesick/cli.rb:151:in `block (2 levels) in link'
        from /usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/lib/homesick/cli.rb:150:in `each'
        from /usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/lib/homesick/cli.rb:150:in `block in link'
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions.rb:184:in `block in inside'
        from /usr/lib/ruby/2.2.0/fileutils.rb:127:in `chdir'
        from /usr/lib/ruby/2.2.0/fileutils.rb:127:in `cd'
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/actions.rb:184:in `inside'
        from /usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/lib/homesick/cli.rb:143:in `link'
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
        from /usr/lib/ruby/gems/2.2.0/gems/homesick-1.1.2/bin/homesick:9:in `<top (required)>'
        from /usr/bin/homesick:23:in `load'
        from /usr/bin/homesick:23:in `<main>'
```

System info:
```
$ uname -a
Linux athena.barrera.io 3.19.0-1-ARCH #1 SMP PREEMPT Mon Feb 9 07:08:20 CET 2015 x86_64 GNU/Linux
$ ruby --version
ruby 2.2.1p85 (2015-02-26 revision 49769) [x86_64-linux]
```

**Comment by ZachBeta** [comment]
Running into the same issue
```
╭ zach@Zach-MBA  ~/.homesick/repos/dotfiles ‹ruby-2.2.2@global› ‹master*›
╰ $ homesick symlink
   identical  /Users/zach/.bash
    conflict  /Users/zach/.bashrc exists
Overwrite /Users/zach/.bashrc? (enter "h" for help) [Ynaqdh] d
/Users/zach/.rvm/gems/ruby-2.2.2@global/gems/homesick-1.1.2/lib/homesick/utils.rb:153:in `block in collision_accepted?': undefined local variable or method `source' for #<Homesick::CLI:0x007f9adb1c7988> (NameError)
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/thor-0.19.1/lib/thor/shell/basic.rb:254:in `block in file_collision'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/thor-0.19.1/lib/thor/shell/basic.rb:237:in `loop'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/thor-0.19.1/lib/thor/shell/basic.rb:237:in `file_collision'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/homesick-1.1.2/lib/homesick/utils.rb:153:in `collision_accepted?'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/homesick-1.1.2/lib/homesick/actions/file_actions.rb:78:in `handle_symlink_action'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/homesick-1.1.2/lib/homesick/actions/file_actions.rb:61:in `ln_s'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/homesick-1.1.2/lib/homesick/utils.rb:201:in `block in symlink_each'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/homesick-1.1.2/lib/homesick/utils.rb:188:in `block (2 levels) in each_file'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/homesick-1.1.2/lib/homesick/utils.rb:162:in `each'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/homesick-1.1.2/lib/homesick/utils.rb:162:in `block in each_file'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/thor-0.19.1/lib/thor/actions.rb:184:in `block in inside'
	from /Users/zach/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/fileutils.rb:127:in `chdir'
	from /Users/zach/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/fileutils.rb:127:in `cd'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/thor-0.19.1/lib/thor/actions.rb:184:in `inside'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/homesick-1.1.2/lib/homesick/utils.rb:158:in `each_file'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/homesick-1.1.2/lib/homesick/utils.rb:200:in `symlink_each'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/homesick-1.1.2/lib/homesick/cli.rb:147:in `block in link'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/thor-0.19.1/lib/thor/actions.rb:184:in `block in inside'
	from /Users/zach/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/fileutils.rb:127:in `chdir'
	from /Users/zach/.rvm/rubies/ruby-2.2.2/lib/ruby/2.2.0/fileutils.rb:127:in `cd'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/thor-0.19.1/lib/thor/actions.rb:184:in `inside'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/homesick-1.1.2/lib/homesick/cli.rb:143:in `link'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/gems/homesick-1.1.2/bin/homesick:9:in `<top (required)>'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/bin/homesick:23:in `load'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/bin/homesick:23:in `<main>'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/bin/ruby_executable_hooks:15:in `eval'
	from /Users/zach/.rvm/gems/ruby-2.2.2@global/bin/ruby_executable_hooks:15:in `<main>'
╭ zach@Zach-MBA  ~/.homesick/repos/dotfiles ‹ruby-2.2.2@global› ‹master*›
╰ $ homesick --version
1.1.2
```

**Comment by feoh** [comment]
+1 on this.  OSX Yosemite, Ruby 2.1.5p273 (Bundled with ChefDK)

**Comment by JCook21** [comment]
I've finally had a chance to look at this and I think I have a fix at https://github.com/JCook21/homesick/commit/75bf94dae51fb4ccf4a04aea7c95b0c36f61a3b6. @gerasiov, @hobarrera, @ZachBeta, @feoh would any of you like to verify this works?

**Comment by hobarrera** [comment]
Fails if the file is a directory:

```
    conflict  /home/hugo/.config/todoman exists
Overwrite /home/hugo/.config/todoman? (enter "h" for help) [Ynaqdh] d
/usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/shell/color.rb:108:in `binread': Is a directory @ io_fread - /home/hugo/.confi
g/todoman (Errno::EISDIR)
```

If it's a regular file, diff compares the contents of one, with the path of the other:

```
    conflict  /home/hugo/.purple/prefs.xml exists
Overwrite /home/hugo/.purple/prefs.xml? (enter "h" for help) [Ynaqdh] d
- <?xml version='1.0' encoding='UTF-8' ?>
+ /home/hugo/.homesick/repos/dotfiles/home/.purple/prefs.xml
- (the rest of the contents of the file...)
```

**Comment by JCook21** [comment]
@hobarrera I've had another crack at this in https://github.com/JCook21/homesick/commit/1f015480c7b2718718428a6cf3e797bbc574c587. Could you take a look? I'm not sure about the best testing approach for this since the diffing functionality is coming from Thor. That said, we're having to hack the base Thor::Shell::Basic::show_diff method to support diffing symlinks and now checking for directories so maybe it is time for a test or two on this.

**Comment by JCook21** [comment]
I've merged #146 now and will try to cut a new release with this in it over the next few days.
